src/lib.rs

```diff
@@ -275,7 +339,12 @@ pub mod logs {
             request: tonic::Request<base::ExportLogsServiceRequest>,
         ) -> Result<tonic::Response<base::ExportLogsServiceResponse>, tonic::Status> {
             match self.channel.send(request.into_inner()).await {
-                Ok(_) => Ok(tonic::Response::new(base::ExportLogsServiceResponse {})),
+                Ok(()) => Ok(tonic::Response::new(base::ExportLogsServiceResponse {
+                    partial_success: Some(base::ExportLogsPartialSuccess {
+                        rejected_log_records: 0,
+                        error_message: "snot".to_string(),
```

Comment:
We probably don't want `"snot"`'s in non test code ;)

same bellow

also, should we return `None` here it's from the rest of the code I got the impression that `partial_success` implies some kind of failure and not a complete success, or do I misunderstand it's meaning?

Reply:
I clearly had lunch and completely forgot that one 👍

Reply:
I have the same concern with `partial` - but we could get `rejected == 0` and a non-zero error message. There's no easy way to determine the sentiment of that error_message as success `"OK"` or failure `"Eaten by dragons"`. I would have expected a set of trace/span ids instead of a count but perhaps a future version of otel will make the now fallible responses somewhat useful other than for metrics capture `We failed N records but we're not telling you why in a structured and useful way` is what we got today!

Reply:
I'm going to let the `partial` hang wet as this is something that will likely converge to a normative behaviour by OpenTelemetry implementations - and that will suggest how resolution by de facto standard 🤷  - the `snot` however will be whacked on the next push!

Reply:
Lol, not a super creative fix:

```rust
partial_success: Some(base::ExportLogsPartialSuccess {
                        rejected_log_records: 0,
                        error_message: "Ok".to_string(),
                        ...
```

Reply:
Donezo - this nose had 3 snotty nostrils it seems!